File: BE/src/main/java/com/ssafy/e106/domain/subscriptionusage/dto/response/SubscriptionUsageReportItemResponse.java
package com.ssafy.e106.domain.subscriptionusage.dto.response;

import java.time.LocalDate;

public record SubscriptionUsageReportItemResponse(
    Long subscriptionId,
    String subscriptionType,
    Long serviceId,
    String bundleCode,
    String serviceName,
    String planName,
    String category,
    String logoUrl,
    Integer monthlyPrice,
    Integer totalUsedMinutes,
    Integer usedDays,
    LocalDate lastUsedDate,
    Integer hourlyCost,
    String nudgeMessage) {
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscriptionusage/SubscriptionUsageReportItemResponse.kt
package com.ssafy.e106.data.dto.subscriptionusage

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionUsageReportItemResponse(
    val subscriptionId: Long,
    val subscriptionType: String,
    val serviceId: Long? = null,
    val bundleCode: String? = null,
    val serviceName: String,
    val planName: String,
    val category: String? = null,
    val logoUrl: String? = null,
    val monthlyPrice: Int,
    val totalUsedMinutes: Int,
    val usedDays: Int,
    val lastUsedDate: String? = null,
    val hourlyCost: Int? = null,
    val nudgeMessage: String? = null,
)
